File: backend/src/Domain/Biblioteca.Domain/Livro.cs
namespace Biblioteca.Domain
{
    public class Livro
    {
        public enum StatusLivro // enum com o status do livro 
        {
            Disponível,
            Emprestado,
            Reservado,
        }
        public int Id { get; set; }  // o que o meeu bd vai receber dda entidade livro 
        public string ISBN { get; set; } = string.Empty;
        public string Titulo { get; set; } = string.Empty;
        public int Ano { get; set; }
        public string Autor { get; set; } = string.Empty;

        public StatusLivro Status { get; set; }

    }
}

File: frontend/src/types/interfaces/livro.ts
export type StatusLivroApi = 0 | 1 | 2

export type StatusLivro = 'Disponível' | 'Emprestado' | 'Reservado'

export interface LivroApi {
  id: number
  isbn: string
  titulo: string
  autor: string
  ano: number
  status: StatusLivroApi
}

export interface Livro {
  id: number
  isbn: string
  titulo: string
  autor: string
  ano: number
  status: StatusLivro
}

export const statusLivroPorCodigo: Record<StatusLivroApi, StatusLivro> = {
  0: 'Disponível',
  1: 'Emprestado',
  2: 'Reservado',
}